### PR TITLE
Make sure info block for external links has at least 3 bytes

### PR DESCRIPTION
**Pull request by e4t**
According to the specification, the information block for external links contains 1 byte of version/flag information and two 0 terminated strings for the object linked to and the full path.
Although not very useful, the minimum string length for each (with terminating 0) would be one byte.
Checking this will help to avoid SEGVs triggered by bogus files.

This fixes CVE-2018-16438 / Bug #2233.

Signed-off-by: Egbert Eich <eich@suse.com>

### CODE REVIEW

**Review by hyoklee — APPROVED** [review]
@e4t , thank you again for your PR!

File: src/H5Olink.c [hunk 1]
```diff
@@ -239,6 +239,8 @@ H5O__link_decode(H5F_t *f, H5O_t H5_ATTR_UNUSED *open_oh, unsigned H5_ATTR_UNUSE
 
             /* A UD link.  Get the user-supplied data */
             UINT16DECODE(p, len)
+            if (lnk->type == H5L_TYPE_EXTERNAL && len < 3)
+                HGOTO_ERROR(H5E_OHDR, H5E_OVERFLOW, NULL, "external link information lenght < 3")
```
Comment (hyoklee) [inline src/H5Olink.c]: @e4t , please fix typo: `lenght` -> length

**Review by jhendersonHDF — APPROVED** [review]

File: release_docs/RELEASE.txt [hunk 2]
```diff
@@ -149,8 +149,17 @@ Bug Fixes since HDF5-1.13.3 release
 ===================================
     Library
     -------
-    -
- 
+    -  Fix CVE-2018-16438 / GHSA-9xmm-cpf8-rgmx / Bug #2233
```
Comment (lrknox) [inline release_docs/RELEASE.txt]:
```suggestion
    -  Fix CVE-2018-16438 / GHSA-9xmm-cpf8-rgmx
```

File: release_docs/RELEASE.txt [hunk 3]
```diff
@@ -149,8 +149,17 @@ Bug Fixes since HDF5-1.13.3 release
...
+       Make sure info block for external links has at least 3 bytes.
+    
+       According to the specification, the information block for external links
+       contains 1 byte of version/flag information and two 0 terminated strings
+       for the object linked to and the full path.
+       Although not very useful, the minimum string length for each (with
+       terminating 0) would be one byte.
+       Checking this helps to avoid SEGVs triggered by bogus files.
+
+       (EFE - 2022/10/09 GH-2234)
```
Comment (lrknox) [inline release_docs/RELEASE.txt]:
```suggestion
       (EFE - 2022/10/09 GH-2233)
```

**Review by lrknox — APPROVED** [review]

**Pull request merged**